pkg/reconciler/instances/istio/action.go

```diff
@@ -21,27 +20,16 @@ const (
 
 type bootstrapIstioPerformer func(logger *zap.SugaredLogger) (actions.IstioPerformer, error)
 
-type ReconcileAction struct {
-	// Temporary solution to overcome Reconciler limitation: Unable to bootstrap IstioPerformer only once in the component reconciler lifetime
-	getIstioPerformer bootstrapIstioPerformer
-}
-
-// NewReconcileAction returns an instance of ReconcileAction
-func NewReconcileAction(getIstioPerformer bootstrapIstioPerformer) *ReconcileAction {
-	return &ReconcileAction{getIstioPerformer}
-}
-
-type UninstallAction struct {
+type PreReconcileAction struct {
```

Comment:
You renamed post actions in pretty cool format, e.x. `MutatingWebhookPostAction`. What about sticking to this convention here? How about naming this action: `SanityChecksPreAction` or `IstioChecksPreAction` or somewhat similar?